plugins/poll/lib/irv.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+class DiscoursePoll::Irv
+  def self.irv_outcome(poll_id)
```

Comment:
@jjaffeux @gschlager one lightweight change here might be to wrap this in a Cache with perhaps a minute expiry ... that would at least reduce a significant amount of burden and still users would get an update most of the time?

Reply:
Could/should this be done in SQL instead of in Ruby-land? 🤔

Reply:
Actually, I don't think it is as scary as it looks.

I ran the following code locally and it returned in a split second:

```
require 'securerandom'
require 'faker'

def generate_options(num_options=10)
  options = []

  num_options.times do
    digest_number = SecureRandom.hex(8)  # Generates a random 16-character hex string
    name = Faker::Name.first_name        # Generates a random English first name
    options << { "id": digest_number, "html": name }
  end

  options
end

def generate_random_ballots(x, options)
  ballots = []
  num_candidates = options.size

  x.times do
    ballot = options.shuffle.take(rand(0..num_candidates)).map {|x| x[:id] }
    ballots << ballot
  end

  ballots
end

options = generate_options(50)

random_ballots = generate_random_ballots(15000, options)

 DiscoursePoll::Irv.run_irv(random_ballots, options)
```

Admittedly this was running locally on a 32GB, 8 core machine, but I still think we might be worrying too much here.

So perhaps a good approach here is to somehow communicate this is experimental and if there is feedback about the performance being poor on some configurations, we can iterate the solution.

Reply:
It might make sense to put the poll build IRV option behind an "experimental" site setting?

Reply:
TBH, I tire of site settings 😅 Building polls is such an advanced feature that I don't think it's going to matter anyways. Let's keep it that way and we'll iterate if there are performance issues.